eth/downloader/statesync.go

```diff
@@ -132,7 +132,10 @@ func (d *Downloader) runStateSync(s *stateSync) *stateSync {
 
 		// Send the next finished request to the current sync:
 		case deliverReqCh <- deliverReq:
-			finished = append(finished[:0], finished[1:]...)
+			// Shift out the first request, but also set the emptied slot to nil for GC
+			copy(finished, finished[1:])
+			finished[len(finished)-1] = nil
+			finished = finished[:len(finished)-1]
```

Comment:
You can get the same effect using

```go
finished = append(finished[:0:len(finished)-1], finished[1:]...)
```

Reply:
Actually, this full format seems to reduce the capacity too, meaning it would be a reallocation of the underlying buffer. Am I missing something?

Reply:
I suggested this because I thought you want to avoid growing the slice indefinitely. The threeway slice expression does this. 

If all you want is setting the unused element to nil, your solution is OK.

It's hard to decide on this, but I want to keep the code compact. This operation isn't worth three lines + comment.